### PR TITLE
fix/SOC-6021:Problem with links in notifications mail

### PR DESCRIPTION
**Pull request by lbengayed**

### CODE REVIEW

File: component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java [hunk 1]
```diff
@@ -362,7 +362,7 @@ public Response redirectUrl(@Context UriInfo uriInfo,
       URL_TYPE urlType = URL_TYPE.valueOf(type);
       switch (urlType) {
         case view_full_activity: {
-          targetURL = hasLoggedIn ? Util.getBaseUrl() + LinkProvider.getRedirectUri(ACTIVITY_ID_PREFIX + "?id=" + objectId) : redirectLink;
+          targetURL =  Util.getBaseUrl() + LinkProvider.getRedirectUri(ACTIVITY_ID_PREFIX + "?id=" + objectId) ;
```
Comment (trongtt) [inline component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: Should we also fix for the cases of _reply_activity_highlight_comment_ and _reply_activity_highlight_comment_ ?

BTW, please remove a redundant space.

File: component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java [hunk 2]
```diff
@@ -362,41 +362,41 @@ public Response redirectUrl(@Context UriInfo uriInfo,
       URL_TYPE urlType = URL_TYPE.valueOf(type);
       switch (urlType) {
         case view_full_activity: {
-          targetURL = hasLoggedIn ? Util.getBaseUrl() + LinkProvider.getRedirectUri(ACTIVITY_ID_PREFIX + "?id=" + objectId) : redirectLink;
+          targetURL = Util.getBaseUrl() + LinkProvider.getRedirectUri(ACTIVITY_ID_PREFIX + "?id=" + objectId) ;
```
Comment (thomasdelhomenie) [inline component/service/src/main/java/org/exoplatform/social/service/rest/NotificationsRestService.java]: The variable redirectLink is not used anymore with this change. If it is really not needed, it should be removed.
But did you sse the comment

> //If user is not authenticated, the query parameter will be removed after login
//so we will not redirect to an activity with query parameter but with path parameter
//this new link will be processed on activity stream portlet

?
Did you check that it still works fine and that the redirection redirects to the right activity ?

**Comment by lbengayed** [comment]
@thomasdelhomenie yes,the variable redirectLink is not needed, it should be removed.
Yes, i check that without redirectLink, it still works fine and that the redirection redirects to the right activity.
I think that we don't need the test  hasLoggedIn, because we have "/portal/" in the url targetURL.

**Comment by thomasdelhomenie** [comment]
@lbengayed ok so let's remove it :wink:

**Comment by lbengayed** [comment]
ok

**Pull request merged**